doc/extensibility/dataflow_example.md

```diff
@@ -0,0 +1,168 @@
+# DataFlow Examples
+
+## Original Data Source
+
+``` Csharp
+/// <summary>
+/// An example of an original datasource.
+/// The goal is to listen to an original source and publish a new value whenever it changes.
+/// </summary>
+[Export]
+private class MyOriginalDataSource : ProjectValueDataSourceBase<string>
+{
```

Comment:
It is interesting to see more efficient way to include a sample code in our document.  Should we include a .cs file (instead of a .md file), so someone can copy into their project system to give a try?